components/mdns/mdns.c

```diff
@@ -569,7 +569,7 @@ static const uint8_t * _mdns_read_fqdn(const uint8_t * packet, const uint8_t * s
                     && (strcmp(buf, MDNS_DEFAULT_DOMAIN) != 0)
                     && (strcmp(buf, "ip6") != 0)
                     && (strcmp(buf, "in-addr") != 0)) {
-                snprintf((char*)name, MDNS_NAME_BUF_LEN, "%s.%s", name->host, buf);
+                snprintf((char*)name, MDNS_NAME_BUF_LEN+1, "%s.%s", name->host, buf);
```

Comment:
I don't think this is correct, the destination is only MDNS_NAME_BUF_LEN size.

Reply:
I ran into a version of this one: https://bugzilla.redhat.com/show_bug.cgi?id=1431678

How would you express/fix it?

Reply:
Maybe using strncat instead?

Reply:
I'm actually having trouble reproducing this with my version of gcc 7.2 for some reason. But I don't have a lot of time to spend on their either.

The gcc changelog suggests that the other approach is to check the return value to make sure the output wasn't truncated. This seems like the correct fix:
https://gcc.gnu.org/gcc-7/changes.html (search for -Wformat-truncated)

Reply:
That could work but it is essentially copying name->host on top of itself instead of appending

Reply:
ah right, I missed that this behaviour was appending in place. strlcat() may make the most sense then.

Reply:
@projectgus @negativekelvin Do you guys mean sth like this?:

```
-                snprintf((char*)name, MDNS_NAME_BUF_LEN+1, "%s.%s", name->host, buf);
+              //snprintf((char*)name, MDNS_NAME_BUF_LEN, "%s.%s", name->host, buf);
+                strlcat(name->host, (char*)'.', sizeof(name->host));
+                strlcat(name->host, buf, sizeof(name->host));
```

Here's a good read from Theo: https://www.sudo.ws/todd/papers/strlcpy.html

I hope I applied it correctly for now :-S... is the check for truncation necessary in this case?

Reply:
Yes, like this looks good to me.